docs/guides/migrations/guide.rst

```diff
@@ -0,0 +1,1298 @@
+
+will have a file name a full 116 characters in length.
+
+The CLI output when using squash along with a fixup is pretty informative 
+on its own, so let's just walk through the output as you'll see it 
+in practice.
+
+First we'll begin with this schema:
+
+.. code-block:: sdl
+
+  type User {
+    name: str;
+  }
+
+Then remove ``name: str;`` from the ``User`` type, migrate, put it 
+back again, and migrate. You'll now have multiple files in your
+``/migrations`` folder and will now be able to squash them.
+
+Then change to this schema without migrating it:
+
+.. code-block:: sdl
+
+  type User {
+    name: str;
+    nickname: str;
+  }
+
+Upon using ``edgedb migration create --squash``, the output is first 
+the same as with our previous squash:
+
+.. code-block:: bash
+
+    $ edgedb migration create --squash
+    Current database revision: 
+    m16awk2tzhtbupjrzoc4fikgw5okxpfnaazupb6rxudxwin2qfgy5q
+    While squashing migrations is non-destructive, 
+    it may lead to manual work if done incorrectly.
+
+    Items to check before using --squash:
+    1. Ensure that `./dbschema` dir is comitted
+    2. Ensure that other users of the database have the revision
+    above or can create database from scratch.
+        To check a specific instance, run:
+        edgedb -I <name> migration log --from-db --newest-first --limit 1
+    3. Merge version control branches that contain schema changes 
+    if possible.
+
+    Proceed? [y/n]
+    > y
+
+But after typing ``y``, the CLI will notice that the existing schema 
+differs from what you have and offers to make a fixup file:
+
+.. code-block::
+
+    Your schema differs from the last revision. 
+    A fixup file can be created
+    to automate upgrading other instances to a squashed revision.
+    This starts the usual migration creation process.
+
+    Feel free to skip this step if you don't have
+    other instances to migrate
+
+    Create a fixup file? [y/n]
+    > y
+
+You will then see the the same questions that otherwise show up in 
+a standard migration:
+
+.. code-block::
+
+    db> did you create property 'nickname' of object type 'default::User'? 
+    [y,n,l,c,b,s,q,?]
+    > y
+    Squash is complete.
+
+Finally, the CLI will give some advice on recommended commands when 
+working with git after doing a squash with a fixup.
+
+.. code-block::
+
+    Remember to commit the `dbschema` directory including deleted files
+    and `fixups` subdirectory. Recommended command:
+        git add dbschema
+
+    The normal migration process will update your migration history:
+        edgedb migrate
+
+    $ edgedb migrate
+    
+    Applied m1v3vqmwif4ml3ucbzi555mjgm4myxs2husqemopo2sz2m7otr22ka 
+    (m16awk2tzhtbupjrzoc4fikgw5okxpfnaazupb6rxudxwin2qfgy5q-
+    m1oih6aevfcftysukvofwuth2bsuj5aahkdnpabscry7p7ljkgbxma.edgeql)
+
+
+.. note::
+
+    Squashing is still limited to schema changes, so queries inside 
+    data migrations will be discarded during a squash.
+
+EdgeDB Watch
```

Comment:
I'm not sure how I feel about sorta "branding" the `edgedb watch` command by title casing it. I think I lean toward just referring to it as the command since it only ever exists in that context. If there were also an interface to it via the UI, I could see this making sense, but since it's exclusively that command, I'd lean toward just referring to it as a command.

Reply:
On the subject in general: I've seen EdgeDB Projects in a few areas but agreed that lowercasing is better.

Now with EdgeDB Watch there's a bit of a parsing annoyance for the reader as it's a verb, and also a persistent process that you can refer to.

* You will see EdgeDB watch quickly display
* If you are curious what is happening as EdgeDB watch does its thing, 
* EdgeDB watch will let you know
* EdgeDB watch is best run in a separate instance
* Then EdgeDB watch will suddenly pipe up
* those generated by EdgeDB watch will have a ``MigrationGeneratedBy.DevMode``
* Once you are satisfied with your changes while running EdgeDB watch
* that were made by EdgeDB watch.

A lot of them look for an instant like someone has mistyped "EdgeDB watches" or started typing watch and then opted for another verb.

So now I think I'm leaning towards keeping the capitalization, just because of how easy it is to refer to it and because otherwise you have to rewrite as either the command or "the process created by edgedb watch" (long) or "the process" (vague...what process was that again?).

Reply:
I think I would prefer styling as inline code. It would help with the parsing issue but without having to "brand" the feature.

> You will see `edgedb watch` quickly display

> If you are curious what is happening as `edgedb watch` does its thing…

etc.

It might not work at the beginning of headings or inside links (due to RST quirks), but those instances are rare.
>

Reply:
Seems to work. There are only two capitalized instances left, one in the section title which is fine, and the CLI output which we are just faithfully passing on here. (Maybe small PR later on to change the output for the CLI but in the meantime that's what it shows)

https://github.com/edgedb/edgedb/pull/6656/commits/b50232905c2893ff990d23adc7b21e6ddfbdbaeb